packages/backend-plugin-api/src/services/definitions/SystemMetadataService.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Observable } from '@backstage/types';
+
+/**
+ * @public
+ */
+export interface BackstageInstance {
+  internalUrl: string;
+  externalUrl: string;
+}
+
+/**
+ * @public
+ */
+export interface SystemMetadataService {
+  instances(): Observable<BackstageInstance[]>;
```

Comment:
Hmm, not sure about observable, especially for backend. Another option is `AsyncIterator`, similar to

https://github.com/backstage/backstage/blob/72d019d663cbc19035e82ebecf74b207faacab26/packages/config-loader/src/sources/types.ts#L77

It sucks to produce, but is nice to consume.

Honestly a simple callback might do it in this case though? We're not looking for reporting either completions or errors anyway?

Curious to here why you chose this though. Honestly the main drawback of observables for me is that most people can't help themselvs and end up writing way to complicated code 😅